src/service/config/cli-configuration.ts

```diff
@@ -96,9 +96,11 @@ export class CLIConfiguration extends BaseConfiguration {
    * @returns
    */
   loadToken(): void {
-    if (this.parsedInputs.token) {
-      Container.set(constants.GITHUB.TOKEN, this.parsedInputs.token);
+    if (this.parsedInputs.token && this.parsedInputs.token.length > 0) {
+      Container.set(constants.GITHUB.TOKEN_POOL, this.parsedInputs.token);
+      Container.set(constants.GITHUB.TOKEN, this.parsedInputs.token[0]);
     } else if (process.env.GITHUB_TOKEN) {
+      Container.set(constants.GITHUB.TOKEN_POOL, [process.env.GITHUB_TOKEN]);
       Container.set(constants.GITHUB.TOKEN, process.env.GITHUB_TOKEN);
```

Comment:
I like the explicit `>0` and I can't use it in ternary operator since typescript complains about comparing with undefined `this.parsedInputs.token?.length > 0`
The `>0` just explicitly tells us that we are checking for length and not just whether the field exists or not. If you are okay with it I would like to keep it